flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/taskmanager/TaskManagerLogsHandlerTest.java

```diff
@@ -0,0 +1,102 @@
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for the {@link TaskManagerLogsHandler}.
+ */
+public class TaskManagerLogsHandlerTest extends TestLogger {
+
+	private static final ResourceID EXPECTED_TASK_MANAGER_ID = ResourceID.generate();
+
+	@Test
+	public void testGetTaskManagerLogsList() throws Exception {
+		final TestingResourceManagerGateway resourceManagerGateway = new TestingResourceManagerGateway();
+		final TaskManagerLogsHandler taskManagerLogsHandler = new TaskManagerLogsHandler(
+			() -> CompletableFuture.completedFuture(null),
+			TestingUtils.TIMEOUT(),
+			Collections.emptyMap(),
+			TaskManagerLogsHeaders.getInstance(),
+			() -> CompletableFuture.completedFuture(resourceManagerGateway));
+		final HandlerRequest<EmptyRequestBody, TaskManagerMessageParameters> handlerRequest = createRequest(EXPECTED_TASK_MANAGER_ID);
+		List<LogInfo> logsList = new ArrayList<>();
+		logsList.add(new LogInfo("taskmanager.log", 1024L));
+		logsList.add(new LogInfo("taskmanager.out", 1024L));
+		logsList.add(new LogInfo("taskmanager-2.out", 1024L));
+		resourceManagerGateway.setRequestTaskManagerLogListFunction(EXPECTED_TASK_MANAGER_ID -> CompletableFuture.completedFuture(logsList));
+		LogsInfo logsInfo = taskManagerLogsHandler.handleRequest(handlerRequest, resourceManagerGateway).get();
+		assertEquals(logsInfo.getLogInfos().size(), resourceManagerGateway.requestTaskManagerLogList(EXPECTED_TASK_MANAGER_ID, TestingUtils.TIMEOUT()).get().size());
+	}
+
+	@Test
+	public void testGetTaskManagerLogsListForUnknownTaskExecutorException() throws Exception {
+		final TestingResourceManagerGateway resourceManagerGateway = new TestingResourceManagerGateway();
+		final TaskManagerLogsHandler taskManagerLogsHandler = new TaskManagerLogsHandler(
+			() -> CompletableFuture.completedFuture(null),
+			TestingUtils.TIMEOUT(),
+			Collections.emptyMap(),
+			TaskManagerLogsHeaders.getInstance(),
+			() -> CompletableFuture.completedFuture(resourceManagerGateway));
+		final HandlerRequest<EmptyRequestBody, TaskManagerMessageParameters> handlerRequest = createRequest(EXPECTED_TASK_MANAGER_ID);
+		resourceManagerGateway.setRequestTaskManagerLogListFunction(EXPECTED_TASK_MANAGER_ID -> FutureUtils.completedExceptionally(new UnknownTaskExecutorException(EXPECTED_TASK_MANAGER_ID)));
+		try {
+			taskManagerLogsHandler.handleRequest(handlerRequest, resourceManagerGateway).get();
+		} catch (Exception exception){
+			assertEquals("org.apache.flink.runtime.rest.handler.RestHandlerException: Could not find TaskExecutor " + EXPECTED_TASK_MANAGER_ID + ".", exception.getMessage());
+		}
+	}
+
+	private static HandlerRequest<EmptyRequestBody, TaskManagerMessageParameters> createRequest(ResourceID taskManagerId) throws HandlerRequestException {
+		final Map<String, String> pathParameters = new HashMap<>();
+		pathParameters.put(TaskManagerIdPathParameter.KEY, taskManagerId.toString());
+		final Map<String, List<String>> queryParameters = new HashMap<>();
```

Comment:
Does this have to be mutable? If not:
```
Collections.emptyMap()
```

Reply:
It needn't be mutable, so `Collections.emptyMap()` is ok.